tests/test_rule.py

```diff
@@ -70,3 +70,89 @@ def test_rule_fail():
     )
     result = pytester.runpytest()
     result.assert_outcomes(failed=1)
+
+
+def test_transitive_dependency_succeeds(create_testset):
+    create_testset(
+        ("abcz/__init__.py", ""),
+        ("abcz/moduleA.py", "import abcz.moduleB"),
+        ("abcz/moduleB.py", "import abcz.moduleC"),
+        ("abcz/moduleC.py", "import abcz.moduleD"),
+        ("abcz/moduleD.py", ""),
+    )
+    archrule("rule exclusion").match("abcz.moduleA").should_import("abcz.moduleD").check("abcz")
+
+
+def test_transitive_dependency_via_may_import_succeeds(create_testset):
+    create_testset(
+        ("abcz/__init__.py", ""),
+        ("abcz/moduleA.py", "import abcz.moduleB"),
+        ("abcz/moduleB.py", "import abcz.moduleC"),
+        ("abcz/moduleC.py", "import abcz.moduleD"),
+        ("abcz/moduleD.py", ""),
+    )
+    (
+        archrule("rule exclusion")
+        .match("abcz.moduleA")
+        .may_import("abcz.moduleC")
+        .should_not_import("abcz.moduleD")
+        .check("abcz")
+    )
+
+
+def test_required_transitive_dependency_fails(create_testset, pytester):
+    create_testset(
+        ("abcz/__init__.py", ""),
+        ("abcz/moduleA.py", "import abcz.moduleB"),
+        ("abcz/moduleB.py", "import abcz.moduleC"),
+        ("abcz/moduleC.py", ""),
+    )
+    pytester.makepyfile(
+        """
+        from pytest_archon.plugin import archrule
+        import pytest_archon
+
+        def test_rule_fail():
+            (
+                archrule("rule exclusion")
+                .match("abcz.moduleA")
+                .should_import("abcz.moduleD")
+                .check("abcz")
+            )
+    """
+    )
+    result = pytester.runpytest()
```

Comment:
`pytester.runpytest`, but it makes debugging cumbersome. Did I miss some documentation?

Reply:
What do you mean? Indeed testing is harder. It would be nice if we can rewrite those tests (there are a couple of pytester tests) and check for the accumulated issues. This should be relatively easy in #21, where we have our improved reporting.

Reply:
I'll check. If I don't get it, I'll discuss separately.